pyk/src/tests/integration/proof/test_mini_kevm.py

```diff
@@ -106,3 +164,80 @@ def test_all_path_reachability_prove(
 
             assert proof.status == proof_status
             assert leaf_number(proof) == expected_leaf_number
+
+    @pytest.mark.parametrize(
+        'test_id,spec_file,spec_module,claim_id,max_iterations,max_depth,cut_rules,proof_status,expected_leaf_number',
+        APR_PROVE_TEST_DATA,
+        ids=[test_id for test_id, *_ in APR_PROVE_TEST_DATA],
+    )
+    def test_custom_step(
+        self,
+        kprove: KProve,
+        kcfg_explore: KCFGExplore,
+        test_id: str,
+        spec_file: str,
+        spec_module: str,
+        claim_id: str,
+        max_iterations: int | None,
+        max_depth: int | None,
+        cut_rules: Iterable[str],
+        proof_status: ProofStatus,
+        expected_leaf_number: int,
+        tmp_path_factory: TempPathFactory,
+    ) -> None:
+
+        with tmp_path_factory.mktemp('custom_step_tmp_proofs') as proof_dir:
+            claim = single(
+                kprove.get_claims(
+                    Path(spec_file), spec_module_name=spec_module, claim_labels=[f'{spec_module}.{claim_id}']
+                )
+            )
+            proof1 = APRProof.from_claim(
+                kprove.definition,
+                claim,
+                subproof_ids=[],
+                logs={},
+                proof_dir=proof_dir,
+            )
+
+            proof2 = APRProof.from_claim(
+                kprove.definition,
+                claim,
+                subproof_ids=[],
+                logs={},
+                proof_dir=proof_dir,
+            )
+            new_init_cterm1 = kcfg_explore.cterm_symbolic.assume_defined(proof1.kcfg.node(proof1.init).cterm)
+            proof1.kcfg.let_node(proof1.init, cterm=new_init_cterm1)
+            kcfg_explore.simplify(proof1.kcfg, {})
+
+            prover1 = APRProver(
+                kcfg_explore=kcfg_explore,
+                execute_depth=max_depth,
+                cut_point_rules=[],
+            )
+            prover1.advance_proof(proof1, max_iterations=max_iterations)
+
+            new_init_cterm2 = kcfg_explore.cterm_symbolic.assume_defined(proof1.kcfg.node(proof1.init).cterm)
+            proof2.kcfg.let_node(proof2.init, cterm=new_init_cterm2)
+            kcfg_explore.simplify(proof2.kcfg, {})
+
+            prover2 = APRProver(
+                kcfg_explore=kcfg_explore,
+                execute_depth=max_depth,
+                cut_point_rules=cut_rules,
+            )
+            prover2.advance_proof(proof2, max_iterations=max_iterations)
+
+            kcfg_show = KCFGShow(
+                kprove, node_printer=APRProofNodePrinter(proof2, kprove, full_printer=True, minimize=False)
+            )
+            cfg_lines1 = kcfg_show.show(proof1.kcfg)
+            _LOGGER.info('\n'.join(cfg_lines1))
```

Comment:
I don't understand how we're controlling whether the custom step fires or not. I guess we want to always have the rule as a cut-rule, but in one test use the custom step, and in the other one don't, but make sure we get the same KCFG either way?

Reply:
Yes, I wasn't quite sure how to handle this on Friday and I had some errors that were preventing me from running the test locally. 

Now the rule is always marked as a `cut-rule` and I added a `globalvar skip_custom_test` flag in the `custom_step` that is `False` for the first proof and `True` for the second one, then I compare the kcfgs.